enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,799 @@
+  - The pod placement controller will be supported in High Availability (HA)
+    mode
+  - The webhook will be a stateless component
+  - The pod placement controller will support namespace filtering to skip pods
+    that should not be mutated
+  - No support for customized network scenarios (restricted, proxied,
+    disconnected...)
+  - No caching facilities for the image metadata
+  - No graceful uninstallation support: any remaining gated pods have to be
+    manually patched to resume scheduling after uninstallation
+- Phase 2 - General Availability (GA):
+  - support for customized network scenarios (restricted, proxied,
+    disconnected...)
+  - support mirror registries through ICSPs/IDMSs
+  - implement caching facilities for the image metadata
+- Phase 3:
+  - Future performance optimizations
+
+### Workflow Description
+The following section will highlight the main workflows the operator will be
+responsible for from a high-level user perspective.
+
+#### Operator installation
+1. The cluster-admin submits a Subscription object to install the
+   multiarch-managed-operator, either via CLI or the OperatorHub
+2. The cluster admin submits the default `PodPlacementConfig` Custom Resource
+   (CR) adding their specific configuration related to initializing the
+   multiarch-manager-operator's components; the cluster admin can customize the
+   CR by excluding namespaces through regexes/globs, a list of configmaps for
+   registries that use self signed certs (if any) and customize logging levels
+4. The operator reconciles the CR. Its information supports the
+   `PodPlacementController` controller and the `SchedulingGateMutatingWebhook`
+   mutating admission webhook
+
+#### Pod Admission: gating
+1. The user submits a pod
+2. The mutating admission webhook is queried and decides whether to add the
+   scheduling gate to the pod, based on the namespaces exclusion list
+3. The pod is admitted and persisted
+4. The pod is put on hold by the scheduler until all the scheduling gates are
+   removed
+
+#### NodeAffinity patching of gated pods
+1. The `PodPlacementController`, watching for pod events, gets notified of a pod
+   event
+2. If the multiarch-manager-operator's gate is not present, the pod is ignored
+3. If the gate is present, a set of unique images is built from the pod spec
+4. Each image is inspected to gather the supported architectures set
+5. The intersection of the supported architecture sets builds the pod compatible
+   architecture set
+6. If no inspection errors occurred, the
+   `podSpec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution`
+   stanza is patched with the architecture node selector requirements. If any
+   node selector term is present, they are all patched. Otherwise, a new node
+   selector term is added to the `.matchExpressions` list. The merge strategy
+   described here follows the rules in KEP-3838[^2]
+7. The scheduling gate is removed from the pod spec and the pod is updated
```

Comment:
I can add the concern around leader election latency to the open questions. One way to mitigate this would be to setup the operator manager with ` LeaderElectionReleaseOnCancel` since we are not planning any cleanup (yet)

Reply:
added this concern to the open questions.